src/cryptonote_protocol/cryptonote_protocol_handler.inl

```diff
@@ -2564,6 +2603,14 @@ skip:
   }
   //------------------------------------------------------------------------------------------------------------------------
   template<class t_core>
+  void t_cryptonote_protocol_handler<t_core>::hit_score(cryptonote_connection_context &context, unsigned int score)
+  {
+    context.m_score -= score;
```

Comment:
This does not appear to be safe against overflow, although that possibility seems rather unlikely with current rules. Also, it seems a bit odd to specify widths for the p2p score field, but not for the increments.

Reply:
What do you mean by widths ?

Reply:
By widths, I mean `int32` for the field vs. using a native unsigned int for the increment. If the goal is to clarify the mathematics with explicit integer bit-widths, best to use those everywhere; if the goal is to avoid false dependencies by using native integer types, that works, too; either way, it seems superior to be consistent.